clusterloader2/pkg/prometheus/manifests/prometheus-rules.yaml

```diff
@@ -10,17 +10,17 @@ spec:
   - name: apiserver.rules
     rules:
     - expr: |
-        histogram_quantile(0.99, sum(rate(apiserver_request_duration_seconds_bucket[5m])) by (resource,  subresource, verb, scope, le))
+        histogram_quantile(0.99, sum(rate(apiserver_request_sli_duration_seconds_bucket[5m])) by (resource,  subresource, verb, scope, le))
```

Comment:
@wojtek-t do you perhaps have any suggestions on how we should handle the versioning of the manifests?

The less intrusive option that I can see would be to add a recording rule to select the metric and always use the newest available. e.g:
```
expr: |
  (
    apiserver_request_sli_duration_seconds_bucket
    or
    apiserver_request_slo_duration_seconds_bucket
    or
    apiserver_request_duration_seconds_bucket
  )
```
The problems if we were to do that are:
1. since we don't do any aggregation, the evaluation of this new recording rule could take quite some time
2. freshness will now depend on both the scrape interval and the evaluation interval

Reply:
I'm not super familiar with this, so let me ask some questions:
1) What is this manifest used for?
  [Do I understand correctly that it's effectively an input to prometheus server and then our measurements (SLOs) effectively call these queries (which prometheus server has now somehow precomputed?]

2) Would it be an option that we effectively will have:
- all possible queries listed here in manifest (would have a query on a metric that doesn't exist break something)?
- and just the measurements (SLOs) checking cluster version and based on that chosing the right query?

Reply:
1. Yes, that is exactly how you understood it. This particular manifest seems to be used in a couple a places such as the dashboards and the SLO measurements: https://github.com/kubernetes/perf-tests/blob/master/clusterloader2/pkg/measurement/common/slos/api_responsiveness_prometheus.go#L52

2. If the metric doesn't exist, the query will not yield any result, so as long as we don't use it, nothing will break. We could create recording rules (precomputed queries) for all the metrics and then select in code which one to use depending on the cluster version and that would solve the problem for the measurements.

It might be a bit more challenging for the dashboards though, I will try to see if I can find a query that would make this work.